src/modules/axis.js

```diff
@@ -1,26 +1,40 @@
 import i18n from '@dhis2/d2-i18n'
+import { AXIS_ID_COLUMNS, AXIS_ID_ROWS, AXIS_ID_FILTERS } from './layout/axis'
 import {
-    AXIS_ID_COLUMNS,
-    AXIS_ID_ROWS,
-    AXIS_ID_FILTERS,
-    AXIS_ID_YEAR_OVER_YEAR_SERIES,
-    AXIS_ID_YEAR_OVER_YEAR_CATEGORY,
-} from './layout/axis'
+    LAYOUT_TYPE_DEFAULT,
+    LAYOUT_TYPE_PIE,
+    LAYOUT_TYPE_YEAR_OVER_YEAR,
+    LAYOUT_TYPE_PIVOT_TABLE,
+} from './layoutTypes'
 
-const axisNames = {
-    [AXIS_ID_COLUMNS]: i18n.t('Series'),
-    [AXIS_ID_ROWS]: i18n.t('Category'),
-    [AXIS_ID_FILTERS]: i18n.t('Filter'),
-    [AXIS_ID_YEAR_OVER_YEAR_SERIES]: i18n.t('Series'),
-    [AXIS_ID_YEAR_OVER_YEAR_CATEGORY]: i18n.t('Category'),
+const getAxisNamesByLayoutType = layoutType => {
+    switch (layoutType) {
+        case LAYOUT_TYPE_DEFAULT:
+        case LAYOUT_TYPE_PIE:
+        case LAYOUT_TYPE_YEAR_OVER_YEAR:
+        default:
+            return {
+                [AXIS_ID_COLUMNS]: i18n.t('Series'),
+                [AXIS_ID_ROWS]: i18n.t('Category'),
+                [AXIS_ID_FILTERS]: i18n.t('Filter'),
+            }
+        case LAYOUT_TYPE_PIVOT_TABLE:
+            return {
+                [AXIS_ID_COLUMNS]: i18n.t('Columns'),
+                [AXIS_ID_ROWS]: i18n.t('Rows'),
+                [AXIS_ID_FILTERS]: i18n.t('Filter'),
+            }
+    }
 }
 
-export const getAxisName = axisId => {
-    const name = axisNames[axisId]
-
+export const getAxisNameByLayoutType = (axisId, layoutType) => {
+    const name = getAxisNamesByLayoutType(layoutType)[axisId]
     if (!name) {
         throw new Error(`${axisId} is not a valid axis id`)
```

Comment:
This error should be updated to include the `layoutType`, since it might be a valid `axisId` in a different context:

```js
throw new Error(`${axisId} is not a valid axis id in layout ${layoutType}`) // Assuming this has a meaningful string value here...
```

Reply:
Keep in mind that `layoutType` doesn't have a display name so it would show up as e.g. `X is not a valid axis id in layout LAYOUT_TYPE_PIVOT_TABLE`. This error is really just for developers, if you for whatever reason send in something thats not an actual `axisId`. All axes are represented with a name in all layouts, so the case of an axis being part of one layout but not another isn't valid currently. But I don't see the harm in this 🙂